<img src="https://userweb.fct.unl.pt/~jmc.xavier/MAI/iLogos/logo_novafct.png" width="200">

# Departamento de Engenharia Mecânica e Industrial
## Mecânica Aplicada I 

## ANÁLISE DE TRELIÇAS

### Problema 1 <a name="PI"></a>

A treliça da figura suporta três forças concentradas de 17,79 kN.

[a)](#A) A treliça da figura é uma treliça simples? Justifique

[b)](#B) Classifique interior e exteriormente a treliça.

[c)](#C) Aplicado o método dos nós determine o esforço a que estão sujeitas as barras que a constituem, indicando se se encontram à tração ou à compressão.

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1.jpg"
width="600">


#### Resumo

**Estrutura**

Sistema de elementos ligados, construído para suportar ou transferir forças e para resistir com segurança às cargas que nele atuam. Na análise das forças das estruturas é necessário desmembrar (separar nos nós e barras que o constituem) a estrutura e analisar, separadamente, os diagramas de corpo livre dos elementos individuais ou da combinação dos elementos, de maneira a determinar as forças internas da estrutura.

**Treliça**

Uma treliça é uma estrutura constituída por barras retilíneas ligadas umas às outras por articulações (nós). Portanto, as forças atuantes nas extremidades de uma barra reduzem se a uma única força sem binário.

**Problema de Equilibrio Bidimensional**

**$r$** : número de reações ou ligações ao exterior (do diagrama de corpo livre da treliça as 3 eqs. de equilíbrio: $\sum F_x=0; \sum F_y=0; \sum M_O=0$, são utilizadas para calcular as reações).

**$m$** : número de incognitas que são os esforços nas barras.

**$n$** : número de nós (do diagrama de corpo livre para cada nó têm-se: $\sum F_x=0; \sum F_y=0 ~\Rightarrow~ 2n$ equações =  n.º máximo de equações de equilíbrio independentes).


### Resolução

#### a) A treliça da figura é uma treliça simples? Justifique. <a id="A"></a> ([_enunciatum_](#PI))

Considere-se uma treliça rígida (isto é uma treliça que não entrará em colapso) constituida por três barras ligadas entre si. Uma treliça simples é qualquer treliça que possa ser gradualmente construida a partir desta treliça triangular básica, pela junção sucessiva de duas novas barras, ligadas a diferentes nós existentes e interligadas entre sí num novo nó.

A verificação se uma treliça é simples passa por confirmar se é constituída por triângulos (rígidos) e se é globalmente isostática.


#### b) Classifique interior e exteriormente a treliça. <a id="B"></a> ([_enunciatum_](#PI))

Pode-se constatar que a treliça do problema é constituída por vários triângulos (rígidos) e da análise quanto à sua constituição tem-se:

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1a1.jpg"
width="600">

Desta forma, a treliça contem:

| N.º de nós, $n$ | N.º de barras, $m$ | N.º de reações, $r$ |
| ---------- | ------------- | -------------- |
| 8 | 13 | 3 |

- Da análise da estrutura isostática exterior conclui-se: $r = 3$.
- Da análise da estrutura isostática interior conclui-se: $m = 2n-3$.
- Da análise global da estrutura permite concluir:

\begin{equation*}
g = 2n - (m+r)
\end{equation*}

em que,

\begin{equation*}
\begin{aligned}
g &: \textrm{nº de graus de liberdade} \\
2n &: \textrm{nº máx. de equações independentes} \\
m+r &: \textrm{nº total de incógnitas}
\end{aligned}
\end{equation*}

In [1]:

n, m, r = 8, 13, 3
print(f'2n = {2*n:.0f} | m  = {m:.0f} | r  = {r:.0f}')

def grauhiperstaticidade(nn,mm,rr):
    return 2*nn - (mm+rr)

g = grauhiperstaticidade(n,m,r)
print('g  = ','{:.0f}'.format(g))

2n = 16 | m  = 13 | r  = 3
g  =  0


Conclui-se portanto ($g= 0$) que a estrutura é globalmente isostática. Esta é condição necessária mas não suficiente para que a treliça seja estáticamente determinada.

**Estatia global**

Estatia global da treliça ($\alpha_g$) é calculada de acordo com:

\begin{equation*}
\alpha_g = \alpha_e + \alpha_i
\end{equation*}

em que $\alpha_e$ e $\alpha_i$ são a estatia exterior e estatia interior da treliça calculadas, respectivamente, por:

\begin{equation*}
\alpha_e = 3 × \textrm{( nº de apoios − 1)} − \textrm{nº de libertações exteriores}
\end{equation*}

- Apoios: O número de apoios refere-se ao total de apoios que a treliça tem
- Libertações Exteriores: Este termo refere-se ao número de graus de liberdade que não estão restringidos pelos apoios

\begin{equation*}
\alpha_i = 3 × \textrm{nº de malhas fechadas} − \textrm{nº de ligações interiores}
\end{equation*}

- Malha fechada numa treliça refere-se a uma região da estrutura que é completamente delimitada por barras, formando um circuito fechado. Malhas fechadas considera que cada malha adiciona três graus de liberdade internos.
- Ligações interiores referem-se às barras internas da treliça que conectam os nós entre si.

In [1]:
def estatiaexterior(napoios, nlibertacoes):
    return 3 * ( napoios - 1 ) - nlibertacoes

def estatiainterior(nmalhas,nligainter):
    return  3 * nmalhas - nligainter

npoios_, nlibext_ = 2, 3
alphe = estatiaexterior(npoios_, nlibext_)
print(f'estatia exterior = 3({npoios_} - 1) - {nlibext_} =  {alphe}')
nmalh_, nlibint_ = 6, 1*2 + 2*3 + 3*2 + 4*1 
alphi = estatiainterior(nmalh_,nlibint_)
print(f'estatia interior = 3({nmalh_}) - {nlibint_} = {alphi}')
estatiaglobal = alphe + alphi
print((f'estatia global = {estatiaglobal}'))


estatia exterior = 3(2 - 1) - 3 =  0
estatia interior = 3(6) - 18 = 0
estatia global = 0


#### c) Aplicado o método dos nós determine o esforço a que estão sujeitas as barras que a constituem, indicando se se encontram à tração ou à compressão. <a id="C"></a> ([_enunciatum_](#PI))

Começa-se por desenhar o DCL da treliça.

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1a.jpg"
width="600">

**Equilíbrio da estrutura - determinação das reações**

A análise do equilíbrio da treliça plana conduz ao seguinte sistema de equações lineares:

\begin{equation*}
\begin{cases}
\rightarrow(+)&:&\sum F_x = 0 \\
\uparrow(+)&:&\sum F_y = 0 \\
\circlearrowleft(+)&:&\sum M_A =0
\end{cases}~\Leftrightarrow~
\begin{cases}

R_{Ax} &=~0 \\
R_{Ay} - 3F + R_{Hy} &=~0 \\
4aR_{Hy} - aF - 2aF - 3aF &=~0
\end{cases}~\Leftrightarrow~
\begin{cases}
R_{Ax} = 0 \\
R_{Ay} = \frac{3}{2}F \\
R_{Hy} = \frac{3}{2}F
\end{cases}
\end{equation*}

In [3]:
import sympy as sy
from sympy.solvers import solve

# unit: m, kN
LX, LY, F = 2.74, 3.66, 17.79

rax, ray, rhy, f, a = sy.symbols('rax ray rhy f a')

SFX = rax
print('sum Fx :', SFX)
SFY = ray - 3*f + rhy
print('sum Fy :', SFY)
SMA = 4*a*rhy - a*F - 2*a*F - 3*a*F
print('sum MA :', SMA)
sol = solve({SFX,SFY,SMA},{rax, ray, rhy})

RAx = sol[rax]
print('RAx    = ', RAx,' kN')
NRAx = RAx.subs({(f,F),(a,LX)})
# print('N[RAx] = ', NRAx,' kN')
RAy = sol[ray]
print('RAy    = ', RAy)
NRAy = RAy.subs({(f,F),(a,LX)})
print('N[RAy] = ','{:.2f}'.format(NRAy),' kN')
RHy = sol[rhy]
print('RHy    = ','{:.2f}'.format(RHy),' kN')
NRHy = RHy.subs({(f,F),(a,LX)})
# print('N[RHy] = ','{:.2f}'.format(NRHy),' kN')

sum Fx : rax
sum Fy : -3*f + ray + rhy
sum MA : 4*a*rhy - 106.74*a
RAx    =  0.0  kN
RAy    =  3.0*f - 26.685
N[RAy] =  26.68  kN
RHy    =  26.68  kN


**Simplificações do problema por simetria**

Simplificações do  problema devido à simetria e condições especiais de carregamento. Por simetria da geometria e do carregamento:

\begin{equation*}
\begin{cases}
R_{Ay} = R_{Hy}\\
F_{AC} = F_{GH}
\end{cases}~;~
\begin{cases}
F_{AB} = F_{FH}\\
F_{BC} = F_{FG}
\end{cases}~;~
\begin{cases}
F_{BE} = F_{EF}\\
F_{BD} = F_{DF}
\end{cases}~;~
F_{CE} = F_{EG}
\end{equation*}

Além disso, como não existe qualquer força aplicada no nó $D$, $F_{DE} = 0$, ou seja, a barra DE não está submetida a qualquer esforço. A sua função é apenas a de segurança. Os membros com esforço nulo são usados para aumentar a estabilidade da treliça durante a construção e para fornecer um apoio adicional se o carregamento for alterado.

De notar que existirão membros ou barras com esforço nulo sempre que:

- apenas dois membros formarem um nó da treliça e nenhuma carga externa ou reação de apoio for aplicada no nó;

- três membros formarem um nó da treliça e dois dos membros forem colineares, o terceiro membro será um membro de força zero (se nenhuma força externa for aplicada no nó).

A figura abaixo identifica, através da mesma cor, as barras de igual esforço. A branco, identifica-se a barra livre de qualquer esforço interno (normal).

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1a2_.png"
width="800">

**Escolha do primeiro nó**

Havendo apenas duas equações de equilíbrio disponíveis para estabelecer o equilíbrio do nó (partícula), escolhe-se um nó onde sejam desconhecidas apenas duas forças, e.g. o nó A:

**Nó A**

_Diagrama de corpo livre do nó A_

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1b.jpg"
width="300">

Em conformidade com a terceira lei de Newton que afirma que, as forças de ação e reação entre corpos em contacto têm a mesma intensidade, a mesma linha de acção, e sentidos opostos, forças com sentido para fora dos nós indicam tração das barras sobre esse nó, e que, inversamente, forças que entram no nó, indicam que a barra associadas está à compressão.

_Estudo do equilíbrio estático no nó_

Aplicando as equações de equilíbrio ao nó resulta:

\begin{equation*}
\begin{cases}
\sum F_x = 0 \\
\sum F_y = 0
\end{cases}~\Leftrightarrow~
\begin{cases}
F_{AC} + F_{AB}\cos\alpha = 0 \\
R_{Ay} + F_{AB}\sin\alpha = 0
\end{cases}~\Leftrightarrow~
\begin{cases}
F_{AB} = -\frac{R_{Ay}}{\sin\alpha} \\
F_{AC} = \frac{R_{Ay}}{\tan\alpha}
\end{cases}
~\Leftrightarrow~
\begin{cases}
F_{AB} = -33.33 ~\textrm{kN (C)} \\
F_{AC} = 19.98 ~\textrm{kN (T)}
\end{cases}
\end{equation*}

com: $\alpha = \arctan(b/a) = \arctan(L_Y/L_X)$.

In [4]:
import numpy as np

alpha = np.arctan(LY/LX)
print(f'alpha = {alpha:.3f} radianos = {np.rad2deg(alpha):.1f} graus')

fab, fac, t = sy.symbols('fab fac t')

SFX = fac + fab*sy.cos(t)
print('sum Fx :', SFX)
SFY = ray + fab*sy.sin(t)
print('sum Fy :', SFY)
sol = solve({SFX,SFY},{fab, fac})

FAB = sol[fab]
NFAB = FAB.subs({(t,alpha),(ray,NRAy)})
print(f'FAB    = {FAB} = {NFAB:.2f} kN (C)')
FAC = sol[fac]
NFAC = FAC.subs({(t,alpha),(ray,NRAy)})
print(f'FAC    = {FAC} = {NFAC:.2f} kN (T) ')

alpha = 0.928 radianos = 53.2 graus
sum Fx : fab*cos(t) + fac
sum Fy : fab*sin(t) + ray
FAB    = -ray/sin(t) = -33.33 kN (C)
FAC    = ray*cos(t)/sin(t) = 19.98 kN (T) 


Note-se que as forças que têm sentido contrário ao nó (i.e. saem do nó) indicam que a barra está à tração (T), enquanto que as forças que tomam o sentido do nó (i.e. entram no nó) indicam que a barra está à compressão (C).


**Nó C**

Escolhe-se um outro nó onde se desconheçam, novamente, apenas duas forças. Vai-se agora olhar para o nó C:

_Diagrama de corpo livre do nó C_

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1c.jpg"
width="300">

_Estudo do equilíbrio estático no nó_

\begin{equation*}
\begin{cases}
\sum F_x = 0 \\
\sum F_y = 0
\end{cases}~\Leftrightarrow~
\begin{cases}
-F_{AC} + F_{CE} = 0 \\
F_{CB} - F = 0
\end{cases}
~\Leftrightarrow~
\begin{cases}
F_{CE} = F_{AC} = 19.98~\textrm{kN (T)} \\
F_{CB} = F = 17.79~\textrm{kN (T)}
\end{cases}
\end{equation*}

In [5]:
fcb, fce = sy.symbols('fcb fce')

SFX = -fac + fce
print('sum Fx :', SFX)
SFY = fcb - f
print('sum Fy :', SFY)
sol = solve({SFX,SFY},{fcb, fce})

FCB = sol[fcb]
NFCB = FCB.subs({(f,F),(fac,FAC)})
print(f'FCB    = {FCB} = {NFCB:.2f} kN (C)')
FCE = sol[fce]
NFCE = FCE.subs({(f,F),(fac,NFAC)})
print(f'FCE    = {FCE} = {NFCE:.2f} kN (T)')

sum Fx : -fac + fce
sum Fy : -f + fcb
FCB    = f = 17.79 kN (C)
FCE    = fac = 19.98 kN (T)


**Nó B**

Analogamente para o nó B:

_Diagrama de corpo livre do nó B_

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1d.jpg"
width="300">

\begin{equation*}
\begin{cases}
\sum F_x = 0 \\
\sum F_y = 0
\end{cases}~\Leftrightarrow~
\begin{cases}
F_{BD} - F_{AB}\cos\alpha + F_{BE}\cos\alpha = 0 \\
- F_{AB}\sin\alpha - F_{BE}\sin\alpha - F_{BC}= 0
\end{cases}
\end{equation*}

In [6]:
fbd, fbe = sy.symbols('fbd fbe')

SFX = fbd - fab*sy.cos(t) + fbe*sy.cos(t)
print('sum Fx :', SFX)
SFY = - fab*sy.sin(t) - fbe*sy.sin(t) - fcb
print('sum Fy :', SFY)
sol = solve({SFX,SFY},{fbd, fbe})

FBD = sol[fbd]
NFBD = FBD.subs({(fab,NFAB),(fcb,NFCB),(t,alpha)})
print(f'FBD    = {FBD} = {NFBD:.2f} kN (C)')
FBE = sol[fbe]
NFBE = FBE.subs({(fab,NFAB),(fcb,NFCB),(t,alpha)})
print(f'FBE    =  {FBE} = {NFBE:.2f} kN (T)')

sum Fx : -fab*cos(t) + fbd + fbe*cos(t)
sum Fy : -fab*sin(t) - fbe*sin(t) - fcb
FBD    = 2*fab*cos(t) + fcb*cos(t)/sin(t) = -26.64 kN (C)
FBE    =  -fab - fcb/sin(t) = 11.11 kN (T)


**Nó D**

Escolhe-se um outro nó onde se desconheçam, novamente, apenas duas forças. Analisa-se agora o nó D:

_Diagrama de corpo livre do nó D_

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1e.jpg"
width="300">

\begin{equation*}
\begin{cases}
\sum F_x = 0 \\
\sum F_y = 0
\end{cases}~\Leftrightarrow~
\begin{cases}
F_{DF} = F_{BD} = 26.6~\textrm{kN (C)} \\
F_{DE} = 0
\end{cases}
\end{equation*}

In [7]:
fde, fdf = sy.symbols('fde fdf')

SFX = - fdf + fbd
print('sum Fx :', SFX)
SFY = fde
print('sum Fy :', SFY)
sol = solve({SFX,SFY},{fde, fdf})

FDE = sol[fde]
print('FDE    = ',FDE,' kN')
NFDE = FDE.subs(fbd,NFBD)
FDF = sol[fdf]
NFDF = FDF.subs(fbd,NFBD)
print(f'FBE    = {FBE} = {NFDF:.2f} kN (T)')

sum Fx : fbd - fdf
sum Fy : fde
FDE    =  0  kN
FBE    = -fab - fcb/sin(t) = -26.64 kN (T)


Em resumo, a figura abaixo apresenta os esforços em todas as barras da treliças.

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1f.jpg"
width="600">


### Apêndice. Ftool

<a href="https://www.ftool.com.br/Ftool/"
class="nounderline"
target="_blank" rel="noopener"><span style="color:#333399; font-family: Lucida
Sans; font-size: 9pt; font-style: none; font-weight: bold">  Ftool </a>

<img src="https://userweb.fct.unl.pt/~jmc.xavier//MAI/Notebooks/Ch07/P1/MAI_Ch07_P1g.jpg"
width="600">

### e-Referências

[Understanding and Analysing Trusses](https://www.youtube.com/watch?v=Hn_iozUo9m4)

---

Copyright (c) DEMI - FCT NOVA

Interactive computing by <a href="https://jupyter.org/" target="_blank"> <span
style="color:#333399"> Jupyter Notebook </span> </a> &nbsp;|&nbsp;Coded by <a href = "mailto: jmc.xavier@fct.unl.pt">José Xavier</a>

Licensed under  <a href="http://creativecommons.org/licenses/by-sa/4.0/"
target="_blank"> <span style="color:#333399;font-size: 20px"> CC BY-SA 4.0  </span></a>